pkg/synthetictests/image_pulls.go

```diff
@@ -0,0 +1,76 @@
+package synthetictests
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+)
+
+type eventRecognizerFunc func(event monitorapi.EventInterval) bool
+
+func matchEventForRegexOrDie(regex string) eventRecognizerFunc {
+	regExp := regexp.MustCompile(regex)
+	return func(e monitorapi.EventInterval) bool {
+		return regExp.MatchString(e.Message)
+	}
+}
+
+type singleEventCheckRegex struct {
+	testName      string
+	recognizer    eventRecognizerFunc
+	failThreshold int
+}
+
+// test goes through the events, looks for a match using the s.recognizer function,
+// if a match is found, and occurs more than s.failThreshold times, we mark it as a
+// flake (this allows us to see how often this symptom is happening).
+//
+func (s *singleEventCheckRegex) test(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	success := &junitapi.JUnitTestCase{Name: s.testName}
+	for _, e := range events {
+		if s.recognizer(e) {
```

Comment:
You can actually match multiple events, so you cannot return early. You need to iterate over all of them and categorize into 

```
// messages for events above the fail threshold
failureMessages := []string{}
// messages for events above the flake threshold
flakeMessages := []string{}

// after this loop

if len(fails) > 0{
  return the failure only, with the output as a strings.join of failures and flakes

if len(flakes) >0{
  return the strings.join of flakes and the success like you did before.

otherwise return success
```